Feature/cbv weeks 7 8

mmparton1 opened the pull request:
# .github/pull_request_template.md

## Summary
Implements advanced blog functionality using Django Class-Based Views (CBVs).

## Changes

- Added List, Detail, Create, Update, and Delete views using CBVs.
- Integrated AuthorOrStaffRequiredMixin for secure ownership and access control.
- Implemented pagination and “Prev/Next” navigation.
- Added timezone-aware filtering to hide future posts.
- Updated templates (post_list.html, post_detail.html) with modern layout.
- Cleaned up `.gitignore` and removed cached/venv files.
## How to Test
1) Run the development server:  
   ```bash
   python manage.py runserver

## Risks/Rollback
- Risk: Low/Med/High
- Rollback: `git revert <merge-commit>`

## Links
PRD §<id>, ADR-XXXX, Issue #<id>

# .github/ISSUE_TEMPLATE/feature_request.md

name: Feature request
body:
- type: textarea
  attributes:
    label: Outcome
    description: Link PRD section and describe the user-visible win.
- type: textarea
  attributes:
    label: Acceptance criteria
- type: textarea
  attributes:
    label: Implementation hints / Copilot prompts

mmparton1 committed:
Blog CRUD + auth + Published date/time (auto-fill) + templates

mmparton1 committed:
Update README.MD

Add project deliverables section to top of README.md

mmparton1 committed:
Merge branch 'feature/blog-pr' into FALL2025

mmparton1 committed:
Update README.MD

mmparton1 committed:
Update README.MD

mmparton1 committed:
Fix WSGI configuration and restore manage.py

mmparton1 committed:
Update README.MD

mmparton1 committed:
Add files via upload

mmparton1 committed:
Update README.MD

Update README.md with emoji-enhanced Project Deliverables table and AI Log reference

mmparton1 committed:
Update README.MD

“Updated README with document overview section for clarity”

mmparton1 committed:
Update README.md — complete artifact structure (Module 4 + Blog details)

Updated features and run instructions in README.

mmparton1 committed:
Add AI_LOG.md for Smart Scheduler project documentation for AI - assisted design log+ ethics summary (Moduel 4)

Documented the use of AI tools in the Smart Scheduler project, detailing their purposes, major tasks assisted, reflections on effectiveness, and ethical considerations.

zanehill01 commented:
Hello! This is my submission for Module 4 'Part C. Peer Review (10 points)' - Zane Hill

Your CBV implementation shows strong clarity, maintainability, and correctness. Clarity is evident in the clean use of Django class based views for list, detail, create, update, and delete, explicit template names, and readable slug routing, and it will be even clearer if you add context_object_name on the detail view and pagination on the list view. Maintainability is solid thanks to generic views and a tidy structure that will scale well, and it improves further by adding LoginRequiredMixin to create, update, and delete, enforcing ownership on update and delete with a small UserPassesTestMixin or a queryset scoped to the current author, and centralizing redirects with get_absolute_url so you can remove custom success url methods. Correctness is already good with the list and detail views hiding future dated posts, and it becomes robust once you set the author in form_valid rather than trusting form data, ensure the slug is unique, and allow authors and staff to view their own drafts while keeping them private from others. Overall this is a clear, maintainable, and correct CBV setup that needs only small adjustments to be fully production ready.

mmparton1 committed:
Blog: CBVs + ownership mixin, pagination, templates, auth routes, timezone; clean .gitignore

mmparton1 committed:
Add Week 7–8 Architecture Critique (Part B deliverable in Markdown)

mmparton1 commented:
Added Part B — Week 7–8 Architecture Critique (in /docs/week_7_8_architecture_critique.md)
- Completed CBV refactor and modular design critique.
- All Weeks 7–8 deliverables included and ready for review.

mmparton1 committed:
Add AI_LOG entries for Weeks 5–6 and 7–8 (CBV refactor + Architecture Critique)

mmparton1 committed:
Add Weeks 9–10 Unified Assignment documentation

mmparton1 committed:
Auth + media + blog layout updates for Weeks 9–10

mmparton1 committed:
Merge branch 'FALL2025' of https://github.com/ahuimanu/CIDM6325 into feature/cbv-weeks-7-8

mmparton1 committed:
Replace Weeks 9–10 PDF with Markdown version

mmparton1 committed:
Revise Weeks 9-10 Unified Assignment documentation

Updated the documentation for the Weeks 9-10 Unified Assignment, including project overview, deliverables, and detailed descriptions of implemented features such as Django authentication, admin implementation, and peer review.

ahuimanu commented:
Not accepting PRs.
